In [1]:
import pandas as pd

df = pd.read_csv('../dataset/sonar.csv', header=None)
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 208 entries, 0 to 207
Data columns (total 61 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   0       208 non-null    float64
 1   1       208 non-null    float64
 2   2       208 non-null    float64
 3   3       208 non-null    float64
 4   4       208 non-null    float64
 5   5       208 non-null    float64
 6   6       208 non-null    float64
 7   7       208 non-null    float64
 8   8       208 non-null    float64
 9   9       208 non-null    float64
 10  10      208 non-null    float64
 11  11      208 non-null    float64
 12  12      208 non-null    float64
 13  13      208 non-null    float64
 14  14      208 non-null    float64
 15  15      208 non-null    float64
 16  16      208 non-null    float64
 17  17      208 non-null    float64
 18  18      208 non-null    float64
 19  19      208 non-null    float64
 20  20      208 non-null    float64
 21  21      208 non-null    float64
 22  22

In [2]:
df.head()

0       1       2       3       4       5       6       7       8   \
0  0.0200  0.0371  0.0428  0.0207  0.0954  0.0986  0.1539  0.1601  0.3109   
1  0.0453  0.0523  0.0843  0.0689  0.1183  0.2583  0.2156  0.3481  0.3337   
2  0.0262  0.0582  0.1099  0.1083  0.0974  0.2280  0.2431  0.3771  0.5598   
3  0.0100  0.0171  0.0623  0.0205  0.0205  0.0368  0.1098  0.1276  0.0598   
4  0.0762  0.0666  0.0481  0.0394  0.0590  0.0649  0.1209  0.2467  0.3564   

       9   ...      51      52      53      54      55      56      57  \
0  0.2111  ...  0.0027  0.0065  0.0159  0.0072  0.0167  0.0180  0.0084   
1  0.2872  ...  0.0084  0.0089  0.0048  0.0094  0.0191  0.0140  0.0049   
2  0.6194  ...  0.0232  0.0166  0.0095  0.0180  0.0244  0.0316  0.0164   
3  0.1264  ...  0.0121  0.0036  0.0150  0.0085  0.0073  0.0050  0.0044   
4  0.4459  ...  0.0031  0.0054  0.0105  0.0110  0.0015  0.0072  0.0048   

       58      59  60  
0  0.0090  0.0032   R  
1  0.0052  0.0044   R  
2  0.0095  0.0078   R  
3  0.0040  0.0117   R  
4  0.0107  0.0094   R  

[5 rows x 61 columns]

In [3]:
df.tail()

0       1       2       3       4       5       6       7       8   \
203  0.0187  0.0346  0.0168  0.0177  0.0393  0.1630  0.2028  0.1694  0.2328   
204  0.0323  0.0101  0.0298  0.0564  0.0760  0.0958  0.0990  0.1018  0.1030   
205  0.0522  0.0437  0.0180  0.0292  0.0351  0.1171  0.1257  0.1178  0.1258   
206  0.0303  0.0353  0.0490  0.0608  0.0167  0.1354  0.1465  0.1123  0.1945   
207  0.0260  0.0363  0.0136  0.0272  0.0214  0.0338  0.0655  0.1400  0.1843   

         9   ...      51      52      53      54      55      56      57  \
203  0.2684  ...  0.0116  0.0098  0.0199  0.0033  0.0101  0.0065  0.0115   
204  0.2154  ...  0.0061  0.0093  0.0135  0.0063  0.0063  0.0034  0.0032   
205  0.2529  ...  0.0160  0.0029  0.0051  0.0062  0.0089  0.0140  0.0138   
206  0.2354  ...  0.0086  0.0046  0.0126  0.0036  0.0035  0.0034  0.0079   
207  0.2354  ...  0.0146  0.0129  0.0047  0.0039  0.0061  0.0040  0.0036   

         58      59  60  
203  0.0193  0.0157   M  
204  0.0062  0.0067   M  
205  0.0077  0.0031   M  
206  0.0036  0.0048   M  
207  0.0061  0.0115   M  

[5 rows x 61 columns]

In [17]:
from keras.models import Sequential
from keras.layers.core import Dense
from sklearn.preprocessing import LabelEncoder

import pandas as pd
import numpy
import tensorflow as tf

numpy.random.seed(3)
tf.random.set_seed(3)

df = pd.read_csv('../dataset/sonar.csv', header=None)

dataset = df.values
X = dataset[:,0:60].astype(float)
Y_obj = dataset[:,60]

e = LabelEncoder()
e.fit(Y_obj)
Y = e.transform(Y_obj)

model = Sequential()
model.add(Dense(24, input_dim=60, activation='relu'))
model.add(Dense(10, activation = 'relu'))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])

model.fit(X, Y, epochs=200, batch_size=5)

print("\n Accuracy: %.4f" % (model.evaluate(X, Y)[1]))

Epoch 1/200
42/42 [==============================] - 0s 1ms/step - loss: 0.2432 - accuracy: 0.5865
Epoch 2/200
42/42 [==============================] - 0s 2ms/step - loss: 0.2288 - accuracy: 0.6394
Epoch 3/200
42/42 [==============================] - 0s 2ms/step - loss: 0.2187 - accuracy: 0.6490
Epoch 4/200
42/42 [==============================] - 0s 2ms/step - loss: 0.2031 - accuracy: 0.7260
Epoch 5/200
42/42 [==============================] - 0s 2ms/step - loss: 0.1893 - accuracy: 0.7692
Epoch 6/200
42/42 [==============================] - 0s 1ms/step - loss: 0.1802 - accuracy: 0.7596
Epoch 7/200
42/42 [==============================] - 0s 2ms/step - loss: 0.1677 - accuracy: 0.7740
Epoch 8/200
42/42 [==============================] - 0s 2ms/step - loss: 0.1590 - accuracy: 0.7837
Epoch 9/200
42/42 [==============================] - 0s 2ms/step - loss: 0.1558 - accuracy: 0.7885
Epoch 10/200
42/42 [==============================] - 0s 2ms/step - loss: 0.1483 - accuracy: 0.7788
Epoch 11/

42/42 [==============================] - 0s 2ms/step - loss: 0.0396 - accuracy: 0.9567
Epoch 83/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0357 - accuracy: 0.9712
Epoch 84/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0356 - accuracy: 0.9663
Epoch 85/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0349 - accuracy: 0.9567
Epoch 86/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0347 - accuracy: 0.9712
Epoch 87/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0320 - accuracy: 0.9760
Epoch 88/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0381 - accuracy: 0.9615
Epoch 89/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0312 - accuracy: 0.9760
Epoch 90/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0332 - accuracy: 0.9760
Epoch 91/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0312 - accuracy: 0.9760
Epoch 92/200


42/42 [==============================] - 0s 2ms/step - loss: 0.0099 - accuracy: 0.9952
Epoch 164/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0088 - accuracy: 0.9952
Epoch 165/200
42/42 [==============================] - 0s 2ms/step - loss: 0.0073 - accuracy: 0.9952
Epoch 166/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0073 - accuracy: 0.9952
Epoch 167/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0070 - accuracy: 0.9952
Epoch 168/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0079 - accuracy: 0.9952
Epoch 169/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0068 - accuracy: 0.9952
Epoch 170/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0064 - accuracy: 0.9952
Epoch 171/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0065 - accuracy: 0.9952
Epoch 172/200
42/42 [==============================] - 0s 1ms/step - loss: 0.0063 - accuracy: 0.9952
Epoc

From the above model, we achieved 99.52% of accuracy. However, we are almost sure that this dataset is overfitted. Therefore, we have to split the dataset into train/test dataset using sklearn.model_selection.tran_test_split 

In [22]:
from sklearn.model_selection import train_test_split

from keras.models import Sequential
from keras.layers.core import Dense
from sklearn.preprocessing import LabelEncoder

import pandas as pd
import numpy
import tensorflow as tf

seed = 0
numpy.random.seed(seed)
tf.random.set_seed(3)

df = pd.read_csv('../dataset/sonar.csv', header=None)

dataset = df.values
X = dataset[:,0:60].astype(float)
Y_obj = dataset[:,60]

e = LabelEncoder()
e.fit(Y_obj)
Y = e.transform(Y_obj)

X_train, X_test, Y_train, Y_test = train_test_split(X, Y, test_size=0.3, random_state=seed)

model = Sequential()
model.add(Dense(24, input_dim=60, activation='relu'))
model.add(Dense(10, activation = 'relu'))
model.add(Dense(1, activation='sigmoid'))

model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])

model.fit(X_train, Y_train, epochs=130, batch_size=5)

Epoch 1/130
29/29 [==============================] - 0s 2ms/step - loss: 0.2472 - accuracy: 0.4897
Epoch 2/130
29/29 [==============================] - 0s 2ms/step - loss: 0.2317 - accuracy: 0.6414
Epoch 3/130
29/29 [==============================] - 0s 2ms/step - loss: 0.2241 - accuracy: 0.6207
Epoch 4/130
29/29 [==============================] - 0s 2ms/step - loss: 0.2139 - accuracy: 0.6897
Epoch 5/130
29/29 [==============================] - 0s 2ms/step - loss: 0.2034 - accuracy: 0.7379
Epoch 6/130
29/29 [==============================] - 0s 2ms/step - loss: 0.1925 - accuracy: 0.7517
Epoch 7/130
29/29 [==============================] - 0s 2ms/step - loss: 0.1819 - accuracy: 0.7517
Epoch 8/130
29/29 [==============================] - 0s 2ms/step - loss: 0.1729 - accuracy: 0.8069
Epoch 9/130
29/29 [==============================] - 0s 2ms/step - loss: 0.1650 - accuracy: 0.7862
Epoch 10/130
29/29 [==============================] - 0s 2ms/step - loss: 0.1618 - accuracy: 0.8069
Epoch 11/

29/29 [==============================] - 0s 1ms/step - loss: 0.0107 - accuracy: 1.0000
Epoch 84/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0100 - accuracy: 1.0000
Epoch 85/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0080 - accuracy: 1.0000
Epoch 86/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0073 - accuracy: 1.0000
Epoch 87/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0073 - accuracy: 1.0000
Epoch 88/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0079 - accuracy: 1.0000
Epoch 89/130
29/29 [==============================] - 0s 2ms/step - loss: 0.0067 - accuracy: 1.0000
Epoch 90/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0074 - accuracy: 1.0000
Epoch 91/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0078 - accuracy: 1.0000
Epoch 92/130
29/29 [==============================] - 0s 1ms/step - loss: 0.0065 - accuracy: 1.0000
Epoch 93/130


In [23]:
print("\n Test Accuracy: %.4f" % (model.evaluate(X_test, Y_test)[1]))

2/2 [==============================] - 0s 1ms/step - loss: 0.1447 - accuracy: 0.8571

 Test Accuracy: 0.8571


When we train the dataset (fit the model using train dataset)
and use the model to evaluate the test dataset (evaluate the model using test dataset)
we come up with 85.71% of accuracy

What if we want to save the model we came up with?

In [25]:
from keras.models import load_model

model.save('sonar_model.h5')

In [26]:
model = load_model('sonar_model.h5')

Dividing the dataset into train/test dataset is a good way to avoid overfitting.
However, it still has drawback if the dataset is not enough.
Therefore, we introduce k-fold cross-validation.
StratifiedKFold provides train/test indices to split data in train/test sets.

In [27]:
from sklearn.model_selection import StratifiedKFold

n_fold = 10
skf = StratifiedKFold(n_splits=n_fold, shuffle=True, random_state=seed)

In [35]:
from sklearn.model_selection import train_test_split

from keras.models import Sequential
from keras.layers.core import Dense
from sklearn.preprocessing import LabelEncoder

import pandas as pd
import numpy
import tensorflow as tf

seed = 0
numpy.random.seed(seed)
tf.random.set_seed(3)

df = pd.read_csv('../dataset/sonar.csv', header=None)

dataset = df.values
X = dataset[:,0:60].astype(float)
Y_obj = dataset[:,60]

e = LabelEncoder()
e.fit(Y_obj)
Y = e.transform(Y_obj)

n_fold = 10
skf = StratifiedKFold(n_splits=n_fold, shuffle=True, random_state=seed)

accuracy = []

for train, test in skf.split(X, Y):
    model = Sequential()
    model.add(Dense(24, input_dim=60, activation='relu'))
    model.add(Dense(10, activation = 'relu'))
    model.add(Dense(1, activation='sigmoid'))
    
    model.compile(loss='mean_squared_error', optimizer='adam', metrics=['accuracy'])
    
    model.fit(X[train], Y[train], epochs=130, batch_size=5)
    k_accuracy = "%.4f" %(model.evaluate(X[test], Y[test])[1])
    accuracy.append(k_accuracy)

print("\n %.f fold accuracy:" % n_fold, accuracy)

Epoch 1/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2416 - accuracy: 0.5455
Epoch 2/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2247 - accuracy: 0.6310
Epoch 3/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2116 - accuracy: 0.6578
Epoch 4/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1984 - accuracy: 0.7219
Epoch 5/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1840 - accuracy: 0.7754
Epoch 6/130
38/38 [==============================] - 0s 1ms/step - loss: 0.1699 - accuracy: 0.7861
Epoch 7/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1623 - accuracy: 0.7861
Epoch 8/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1511 - accuracy: 0.8021
Epoch 9/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1432 - accuracy: 0.8021
Epoch 10/130
38/38 [==============================] - 0s 3ms/step - loss: 0.1347 - accuracy: 0.8289
Epoch 11/

38/38 [==============================] - 0s 2ms/step - loss: 0.0118 - accuracy: 1.0000
Epoch 84/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0111 - accuracy: 0.9947
Epoch 85/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0251 - accuracy: 0.9840
Epoch 86/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0236 - accuracy: 0.9786
Epoch 87/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0161 - accuracy: 0.9947
Epoch 88/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0111 - accuracy: 0.9947
Epoch 89/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0098 - accuracy: 1.0000
Epoch 90/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0096 - accuracy: 1.0000
Epoch 91/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0086 - accuracy: 0.9947
Epoch 92/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0079 - accuracy: 1.0000
Epoch 93/130


Epoch 34/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0870 - accuracy: 0.8984
Epoch 35/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0974 - accuracy: 0.8556
Epoch 36/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0877 - accuracy: 0.9091
Epoch 37/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0874 - accuracy: 0.9037
Epoch 38/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0874 - accuracy: 0.8770
Epoch 39/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0857 - accuracy: 0.8984
Epoch 40/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0855 - accuracy: 0.9037
Epoch 41/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0802 - accuracy: 0.9091
Epoch 42/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0795 - accuracy: 0.8984
Epoch 43/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0747 - accuracy: 0.9144


38/38 [==============================] - 0s 1ms/step - loss: 0.0078 - accuracy: 0.9947
Epoch 116/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0083 - accuracy: 1.0000
Epoch 117/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0073 - accuracy: 1.0000
Epoch 118/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0083 - accuracy: 1.0000
Epoch 119/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0080 - accuracy: 0.9947
Epoch 120/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0054 - accuracy: 1.0000
Epoch 121/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0114 - accuracy: 0.9947
Epoch 122/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0065 - accuracy: 1.0000
Epoch 123/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0063 - accuracy: 1.0000
Epoch 124/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0077 - accuracy: 1.0000
Epoc

38/38 [==============================] - 0s 1ms/step - loss: 0.0475 - accuracy: 0.9519
Epoch 67/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0527 - accuracy: 0.9519
Epoch 68/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0455 - accuracy: 0.9519
Epoch 69/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0437 - accuracy: 0.9679
Epoch 70/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0430 - accuracy: 0.9572
Epoch 71/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0436 - accuracy: 0.9679
Epoch 72/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0401 - accuracy: 0.9626
Epoch 73/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0409 - accuracy: 0.9679
Epoch 74/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0416 - accuracy: 0.9679
Epoch 75/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0381 - accuracy: 0.9626
Epoch 76/130


Epoch 17/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1515 - accuracy: 0.8128
Epoch 18/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1492 - accuracy: 0.8021
Epoch 19/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1429 - accuracy: 0.8342
Epoch 20/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1403 - accuracy: 0.8396
Epoch 21/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1376 - accuracy: 0.8075
Epoch 22/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1443 - accuracy: 0.7861
Epoch 23/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1320 - accuracy: 0.8396
Epoch 24/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1350 - accuracy: 0.8449
Epoch 25/130
38/38 [==============================] - 0s 1ms/step - loss: 0.1210 - accuracy: 0.8556
Epoch 26/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1240 - accuracy: 0.8289


38/38 [==============================] - 0s 2ms/step - loss: 0.0243 - accuracy: 0.9840
Epoch 99/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0236 - accuracy: 0.9840
Epoch 100/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0239 - accuracy: 0.9840
Epoch 101/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0214 - accuracy: 0.9840
Epoch 102/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0215 - accuracy: 0.9840
Epoch 103/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0213 - accuracy: 0.9840
Epoch 104/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0202 - accuracy: 0.9840
Epoch 105/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0228 - accuracy: 0.9840
Epoch 106/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0206 - accuracy: 0.9840
Epoch 107/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0212 - accuracy: 0.9840
Epoch

Epoch 49/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0746 - accuracy: 0.9144
Epoch 50/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0764 - accuracy: 0.9305
Epoch 51/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0751 - accuracy: 0.9198
Epoch 52/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0738 - accuracy: 0.9198
Epoch 53/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0703 - accuracy: 0.9305
Epoch 54/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0685 - accuracy: 0.9251
Epoch 55/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0684 - accuracy: 0.9144
Epoch 56/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0779 - accuracy: 0.8984
Epoch 57/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0667 - accuracy: 0.9305
Epoch 58/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0662 - accuracy: 0.9305


1/1 [==============================] - 0s 4ms/step - loss: 0.1764 - accuracy: 0.7619
Epoch 1/130
38/38 [==============================] - 0s 1ms/step - loss: 0.2470 - accuracy: 0.5668
Epoch 2/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2411 - accuracy: 0.5668
Epoch 3/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2354 - accuracy: 0.5668
Epoch 4/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2263 - accuracy: 0.6524
Epoch 5/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2166 - accuracy: 0.6952
Epoch 6/130
38/38 [==============================] - 0s 2ms/step - loss: 0.2071 - accuracy: 0.7273
Epoch 7/130
38/38 [==============================] - 0s 1ms/step - loss: 0.1986 - accuracy: 0.7433
Epoch 8/130
38/38 [==============================] - 0s 1ms/step - loss: 0.1887 - accuracy: 0.7861
Epoch 9/130
38/38 [==============================] - 0s 1ms/step - loss: 0.1782 - accuracy: 0.7701
Epoch 10/130
38/38 [====

38/38 [==============================] - 0s 2ms/step - loss: 0.0625 - accuracy: 0.9412
Epoch 82/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0607 - accuracy: 0.9358
Epoch 83/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0592 - accuracy: 0.9358
Epoch 84/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0575 - accuracy: 0.9572
Epoch 85/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0600 - accuracy: 0.9519
Epoch 86/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0577 - accuracy: 0.9519
Epoch 87/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0557 - accuracy: 0.9572
Epoch 88/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0559 - accuracy: 0.9465
Epoch 89/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0543 - accuracy: 0.9572
Epoch 90/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0526 - accuracy: 0.9572
Epoch 91/130


Epoch 32/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0801 - accuracy: 0.9144
Epoch 33/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0818 - accuracy: 0.9144
Epoch 34/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0745 - accuracy: 0.9198
Epoch 35/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0761 - accuracy: 0.9144
Epoch 36/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0728 - accuracy: 0.9198
Epoch 37/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0699 - accuracy: 0.9305
Epoch 38/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0676 - accuracy: 0.9465
Epoch 39/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0646 - accuracy: 0.9465
Epoch 40/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0649 - accuracy: 0.9305
Epoch 41/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0613 - accuracy: 0.9465


38/38 [==============================] - 0s 2ms/step - loss: 0.0021 - accuracy: 1.0000
Epoch 114/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0022 - accuracy: 1.0000
Epoch 115/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0022 - accuracy: 1.0000
Epoch 116/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0016 - accuracy: 1.0000
Epoch 117/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0015 - accuracy: 1.0000
Epoch 118/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0016 - accuracy: 1.0000
Epoch 119/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0014 - accuracy: 1.0000
Epoch 120/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0017 - accuracy: 1.0000
Epoch 121/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0014 - accuracy: 1.0000
Epoch 122/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0015 - accuracy: 1.0000
Epoc

38/38 [==============================] - 0s 2ms/step - loss: 0.0562 - accuracy: 0.9412
Epoch 65/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0557 - accuracy: 0.9412
Epoch 66/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0534 - accuracy: 0.9519
Epoch 67/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0543 - accuracy: 0.9519
Epoch 68/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0502 - accuracy: 0.9519
Epoch 69/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0493 - accuracy: 0.9572
Epoch 70/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0517 - accuracy: 0.9412
Epoch 71/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0502 - accuracy: 0.9572
Epoch 72/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0465 - accuracy: 0.9572
Epoch 73/130
38/38 [==============================] - 0s 2ms/step - loss: 0.0450 - accuracy: 0.9519
Epoch 74/130


Epoch 15/130
38/38 [==============================] - 0s 1ms/step - loss: 0.1495 - accuracy: 0.8191
Epoch 16/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1450 - accuracy: 0.8138
Epoch 17/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1418 - accuracy: 0.7979
Epoch 18/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1373 - accuracy: 0.8138
Epoch 19/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1403 - accuracy: 0.8085
Epoch 20/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1331 - accuracy: 0.8351
Epoch 21/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1295 - accuracy: 0.8191
Epoch 22/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1311 - accuracy: 0.8085
Epoch 23/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1250 - accuracy: 0.8298
Epoch 24/130
38/38 [==============================] - 0s 2ms/step - loss: 0.1254 - accuracy: 0.8085


38/38 [==============================] - 0s 1ms/step - loss: 0.0314 - accuracy: 0.9734
Epoch 97/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0303 - accuracy: 0.9787
Epoch 98/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0332 - accuracy: 0.9734
Epoch 99/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0291 - accuracy: 0.9734
Epoch 100/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0277 - accuracy: 0.9787
Epoch 101/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0270 - accuracy: 0.9787
Epoch 102/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0264 - accuracy: 0.9840
Epoch 103/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0269 - accuracy: 0.9787
Epoch 104/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0267 - accuracy: 0.9840
Epoch 105/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0337 - accuracy: 0.9628
Epoch 1

Epoch 47/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0723 - accuracy: 0.9202
Epoch 48/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0806 - accuracy: 0.9043
Epoch 49/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0728 - accuracy: 0.9309
Epoch 50/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0716 - accuracy: 0.9362
Epoch 51/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0712 - accuracy: 0.9202
Epoch 52/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0657 - accuracy: 0.9521
Epoch 53/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0642 - accuracy: 0.9255
Epoch 54/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0634 - accuracy: 0.9468
Epoch 55/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0624 - accuracy: 0.9415
Epoch 56/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0610 - accuracy: 0.9415


38/38 [==============================] - 0s 1ms/step - loss: 0.0082 - accuracy: 0.9947
Epoch 129/130
38/38 [==============================] - 0s 1ms/step - loss: 0.0094 - accuracy: 0.9947
Epoch 130/130
1/1 [==============================] - 0s 1ms/step - loss: 0.0958 - accuracy: 0.8500

 10 fold accuracy: ['0.6667', '0.7619', '0.7619', '0.9524', '0.7619', '0.8571', '0.8571', '0.8095', '0.9000', '0.8500']


In [45]:
numpy.mean(list(map(float, accuracy)))

0.81785

In [36]:
from keras.models import load_model

model.save('sonar_model_10_fold.h5')